### PR TITLE
Update thread_safe_store.go, optimize deleteFromIndices method

### PR DESCRIPTION
**Pull request by 249043822**
As all methods of thread_safe_store are threadsafe, so i think, in deleteFromIndices method, if the index is nil, need not run the for structure below, the if shuold position outside

### CODE REVIEW

File: pkg/client/cache/thread_safe_store.go [hunk 1]
```diff
@@ -261,8 +261,8 @@ func (c *threadSafeMap) deleteFromIndices(obj interface{}, key string) error {
 		}
 
 		index := c.indices[name]
-		for _, indexValue := range indexValues {
-			if index != nil {
+		if index != nil {
```
Comment (deads2k) [inline pkg/client/cache/thread_safe_store.go]: We try to keep nesting levels low.  How about
```go
if index == nil{
    continue
}
for ...
```

instead?

Comment (249043822) [inline pkg/client/cache/thread_safe_store.go]: ok, i changed it, thanks, @deads2k

**Comment by 249043822** [comment]
@deads2k,  i have changed it, please test, thanks

**Comment by deads2k** [comment]
lgtm, just needs a squash before merge.

**Comment by 249043822** [comment]
i have squash it in another pr #26888, Thx

**Pull request closed without merging**